File: server/internal/model/location.go
package model

import (
	"github.com/google/uuid"
	"time"
)

type LocationModel struct {
	ID          uuid.UUID `db:"id"`
	Name        string    `db:"name"`
	Description *string   `db:"description"`
	IsDeleted   bool      `db:"is_deleted"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
	IsUser      bool      `db:"is_user"`
}

File: client/src/data/models/location.ts
export type Location = {
  id: string;
  name: string;
  description?: string;
  createdAt: string;
  updatedAt: string;
}

export type CreateLocationRequest = {
  name: string;
  description?: string;
}

export interface TrackableLocation extends Location {
  isUser: boolean;
}